pkg/pillar/docs/nim.md

```diff
@@ -0,0 +1,192 @@
+into the event loop of the manager.
+
+### DpcReconciler
+
+[DpcReconciler](../dpcreconciler/dpcreconciler.go) translates the currently selected device
+port configuration (`DevicePortConfig` struct; abbreviated to DPC) into the corresponding
+low-level network configuration items (routes, IP rules, ARP entries, iptables rules, etc.)
+of the target network stack and applies them using the [Reconciler](../../../libs/reconciler/README.md).
+Internally, DpcReconciler maintains two [dependency graphs](../../../libs/depgraph/README.md),
+one modelling the current state of the network stack and the other the intended state.
+The current state is being updated during each state reconciliation and potentially also when
+a state change notification is received from [NetworkMonitor](#networkmonitor) (e.g. a network
+interface (dis)appearing from/in the host OS). The intended state is rebuilt by DpcReconciler
+based on the input from the DpcManager (selected DPC, intended radio-silence state, etc.).
+This means that in order to understand how a DPC is realized in a network stack (what
+low-level config it maps to), one only needs to look into the DpcReconciler.
+Currently, there is only one implementation of DpcReconciler, created for the Linux network
+stack. To learn what configuration items are used in Linux and how they are represented
+with a dependency graph, see ASCII diagram at the top of [dpcreconciler/linux.go](../dpcreconciler/linux.go)
+file.
+
+### ConnectivityTester
+
+[ConnectivityTester](../conntester/conntester.go) allows to probe the state of external
+connectivity. It exposes only one method `TestConnectivity()`, which should return nil
+error value if the connectivity is working, non-nil value otherwise.
+A special error `RemoteTemporaryFailure` can be returned to indicate that test failed
+due to a remote failure (i.e. the remote endpoint is accessible but fails to respond
+to the test properly). In such case DpcManager can evaluate DPC as valid, hoping that
+the issue with the remote endpoint will be resolved at the other end eventually.
+
+### NetworkMonitor
+
+[NetworkMonitor](../netmonitor/netmonitor.go) allows to:
+
+* list network interfaces present in the network stack
+* obtain (and possibly cache) interface index (aka interface handle)
+* obtain (and possibly cache) interface attributes, addresses, DNS info, etc.
+* watch for interface/address/route/DNS changes
+* clear internal cache to avoid working with stale data
+
+Provided is implementation for Linux network stack based on the netlink interface.
+Also available is a *mock* NetworkMonitor, allowing to simulate a state of a fake
+network stack for the sake of unit-testing of other NIM components.
+
+## Control-flow
+
+The following control-flow diagram shows the sequence of events and the interactions
+between NIM components that follow from a new DPC being published by zedagent and
+received by NIM:
+
+![NIM control-flow](./pics/nim-control-flow.png)
+
+## Debugging
```

Comment:
Is there something useful which can be said about log entries to watch for the operation of NIM.
We have the existing pubsub hooks to log modifications to DNS and DPCList, but are there other structured logs which should be mentioned?

Reply:
Added subsection about logs